### PR TITLE
[Bugfix] NPE in JsonRpc2_0Rx

### PR DESCRIPTION
**Pull request by tau3**
### What does this PR do?
Fixes an NPE (verified by a unit test) 

### Where should the reviewer start?
`JsonRpc2_0RxTest#testReplayBlocksFlowableWhenIOExceptionOnBlockResolving`

### Why is it needed?
fixes bug, simplifies code, improves readability

### CODE REVIEW

File: core/src/main/java/org/web3j/protocol/rx/JsonRpc2_0Rx.java [hunk 1]
```diff
@@ -138,35 +139,21 @@ public JsonRpc2_0Rx(Web3j web3j, ScheduledExecutorService scheduledExecutorServi
     private Flowable<EthBlock> replayBlocksFlowableSync(
             DefaultBlockParameter startBlock,
             DefaultBlockParameter endBlock,
-            boolean fullTransactionObjects,
-            boolean ascending) {
-
-        BigInteger startBlockNumber = null;
-        BigInteger endBlockNumber = null;
+            boolean isFullObjects,
```
Comment (iikirilov) [inline core/src/main/java/org/web3j/protocol/rx/JsonRpc2_0Rx.java:142]: i don't like this name change. maybe `containsFullTransactionObjects` ?

`isFullObjects` is too abstract here

File: core/src/main/java/org/web3j/protocol/rx/JsonRpc2_0Rx.java [hunk 2]
```diff
@@ -138,35 +139,21 @@ public JsonRpc2_0Rx(Web3j web3j, ScheduledExecutorService scheduledExecutorServi
...
-        } else {
-            return Flowables.range(startBlockNumber, endBlockNumber, false)
-                    .flatMap(
-                            i ->
-                                    web3j.ethGetBlockByNumber(
-                                                    new DefaultBlockParameterNumber(i),
-                                                    fullTransactionObjects)
-                                            .flowable());
-        }
+        return Flowables.range(startBlockNumber, endBlockNumber, isAscending)
```
Comment (iikirilov) [inline core/src/main/java/org/web3j/protocol/rx/JsonRpc2_0Rx.java:153]: v. nice :)

Comment (tau3) [reply core/src/main/java/org/web3j/protocol/rx/JsonRpc2_0Rx.java:153]: thank you! 😄

File: core/src/main/java/org/web3j/protocol/rx/JsonRpc2_0Rx.java (hunk 1 above)
Comment (tau3) [reply core/src/main/java/org/web3j/protocol/rx/JsonRpc2_0Rx.java]: yes, i agree)

**Pull request merged**